src/HttpExtractor.php

```diff
@@ -33,6 +33,14 @@ public function __construct(
 
     public function extract(UriInterface $httpSource, string $filesystemDestination): void
     {
+        // check is valid URL
+        if (!preg_match('/^https?/', (string) $httpSource)) {
+            throw new UserException(sprintf(
+                'This url "%s" is not valid',
```

Comment:
`Url "%s" is not valid `. Ale ještě lépe by bylo vytahnout match z toho regexu a vyhodit rovnou `Protocol "%s" is not valid. Only http and https are allowed`. 

Plus ten regex by měl být kompletní IMHO. `/^https?:\/\//`. Takhle je `httpsx://example.com` platný. 

Každopádně ale ten problém byl v tom, že když je "nějak" neplatné URL, tak to vyhodí application error. Takže spíš než validovat to URL by bylo potřeba chytit tu výjimku a vyhodit UserException s odpovídající chybou. 

Kód tam na to už je a předpokládám, že neplatný protokol bude `CURLE_UNSUPPORTED_PROTOCOL`.

Reply:
Případně by ten regex mohl být rovnou v config definition - https://symfony.com/doc/current/components/config/definition.html#validation-rules (ifTrue(function ($v) {regex}) thenInvalid()

Reply:
přehodil jsem ten regex do config definition + udělal na to test